networkx/classes/multidigraph.py

```diff
@@ -564,6 +564,14 @@ def degree_iter(self, nbunch=None, weight=None):
         [(0, 1), (1, 2)]
 
         """
+        if nbunch in self:
+            if weight is None:
+                return sum(len(self.succ[nbunch][node]) for node in self.succ[nbunch]) + \
+                    sum(len(self.pred[nbunch][node]) for node in self.pred[nbunch])
+            s = (sum(sum(self.succ[nbunch][nbr][node].get(weight, 1) for node in self.succ[nbunch][nbr]) for nbr in self.succ[nbunch])) + \
+                (sum(sum(self.pred[nbunch][nbr][node].get(weight, 1) for node in self.pred[nbunch][nbr]) for nbr in self.pred[nbunch]))
+            return s
+
```

Comment:
I would start with ```pred = self.pred[nbunch]```  and ```succ = self.succ[nbunch]```
Then the rest of the code will look very similar to the code just below.

                    indeg = sum([len(data) for data in pred.values()])
                    outdeg = sum([len(data) for data in succ.values()])
                    return (nbunch, indeg + outdeg)